backend/api/v1/project_service.go

```diff
@@ -425,9 +429,119 @@ func (s *ProjectService) SetIamPolicy(ctx context.Context, request *v1pb.SetIamP
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, err.Error())
 	}
+
+	if serviceData, ok := ctx.Value(common.ServiceDataKey).(*anypb.Any); ok {
```

Comment:
This usage doesn't seem to be correct. The method is called before audit processor so we need to set the service data key to the context.